src/main/src/main/java/org/geoserver/ows/CiteComplianceHack.java

```diff
@@ -36,12 +44,66 @@ public boolean preHandle(
             throws Exception {
         if (handler instanceof Dispatcher) {
             Dispatcher dispatcher = (Dispatcher) handler;
-            dispatcher.setCiteCompliant(getInfo().isCiteCompliant());
+            Service service = findService(dispatcher, request, response);
+            if (service != null
+                    && (service.getId().equalsIgnoreCase(getInfo().getId())
+                            || service.getId().equalsIgnoreCase(getInfo().getName()))) {
+                dispatcher.setCiteCompliant(getInfo().isCiteCompliant());
+            }
         }
 
         return true;
     }
 
+    private Service findService(
+            Dispatcher dispatcher, HttpServletRequest request, HttpServletResponse response) {
+        // create a new request instance
+        Request req = new Request();
+
+        // set request / response
+        req.setHttpRequest(request);
+        req.setHttpResponse(response);
+
+        // find the service
+        try {
+            return dispatcher.service(req);
+        } catch (Throwable t) {
```

Comment:
Never catch a Throable, you want Error instances (e..g, OOM) to go free. Catch Exception instead.

Reply:
done